### PR TITLE
Server work

### PR DESCRIPTION
**Pull request by promiseo123**

### CODE REVIEW

File: Webserver/site.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+# In-memory storage for user sessions
+users = {}
+
+
+@app.route('/')
+@app.route('/index')
+def index():
+    if 'username' not in session:
+        return redirect(url_for('/login'))
```
Comment (mjh9585) [inline Webserver/site.py:17]: I was mistaken on these redirects. It is just 'login' and 'index' without the slash.

File: Webserver/site.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    return render_template('index.html', username=session['username'], user_id=session['user_id'])
+
+
+@app.route('/login', methods=['POST'])
+def login():
```
Comment (mjh9585) [inline Webserver/site.py:22]: This route needs to handle both GET and POST.
- GET: render template
- POST: handle login

Please review the project Document

File: Webserver/site.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    return render_template('group.html', username=session['username'], user_id=session['user_id'])
+'''
+
+
+@sock.route('/ws')
+def echo(ws):
+    # emit('/ws', {'username': session['username'], '/ws': msg})
+    while True:
+        data = ws.receive()
+        ws.send(data)
```
Comment (mjh9585) [inline Webserver/site.py:43]: The received message needs to be forwarded to the requested client.

File: Webserver/site.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+from flask import Flask, render_template, request, session, redirect, url_for
+# from flask import SocketIO, emit
+from flask_sock import Sock
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'secret_key'
```
Comment (mjh9585) [inline Webserver/site.py:6]: This should probably be set somewhere else, but for this it is fine. Please assign it a random key value. https://randomkeygen.com/ is a good site for generating these

**Review by mjh9585 — CHANGES_REQUESTED** [review]
I was mistaken about the redirect routes, but overall good start. There are a couple pieces necessary functionality currently missing though. Please review the project document for the detailed requirements.

File: Webserver/site.py (hunk 3 above)
Comment (promiseo123) [reply Webserver/site.py:43]: do you mean that a parameter for a user to forward data to needs to be used?

**Pull request merged**